project/npda/general_functions/csv/csv_upload.py

```diff
@@ -181,6 +181,29 @@ def can_save_field(form, target_field_name):
 
         return True
 
+    # Numbers larger than (max_digits - decimal_places) will fail to save at the database level
+    # https://github.com/rcpch/national-paediatric-diabetes-audit/issues/993
+    def is_too_big_number(model, field_name, value):
+        try:
+            model_definition = model._meta.get_field(field_name)
+
+            max_digits = getattr(model_definition, "max_digits", None)
+            decimal_places = getattr(model_definition, "decimal_places", None)
+
+            if max_digits and decimal_places:
+                max_value = 10 ** (max_digits - decimal_places) - 1
+
+                try:
+                    return value >= max_value
+                # Missing values or strings
+                except TypeError:
```

Comment:
This feels absolutely awful but I guess is in the spirit of 🦆 typing?

Without this the tests that check we handle missing hba1c or passing something totally invalid like "STRING" fail, because the value is either `None` or a string literal at this point and you can't do `str >= number`